src/budget/budgetmanager.cpp

```diff
@@ -1006,8 +1006,31 @@ void CBudgetManager::NewBlock()
         }
     }
 
+    int64_t now = GetTime();
     {
-        // todo: Clean orphan proposal votes and budget finalization votes
+        // Clean orphan proposal votes if no parent arrived after an hour.
+        LOCK(cs_votes);
+        for (auto it = mapOrphanProposalVotes.begin() ; it != mapOrphanProposalVotes.end();) {
+            int64_t lastReceivedVoteTime = it->second.second;
+            if (lastReceivedVoteTime + BUDGET_SYNC_REQUEST_ACCEPTANCE_SECONDS < now) {
+                it = mapOrphanProposalVotes.erase(it);
```

Comment:
should also remove the votes from the mapSeen.

Reply:
yeah, good catch.